v2/zip_longest_test.go

```diff
@@ -0,0 +1,29 @@
+package pie_test
+
+import (
+	"github.com/elliotchance/pie/v2"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestZipLongest(t *testing.T) {
+	for _, test := range zipTests {
+
+		t.Run("", func(t *testing.T) {
+			for i, pair := range pie.ZipLongest(test.Ss1, test.Ss2) {
+				var a int
+				var b float32
+
+				if i < len(test.Ss1) {
+					a = test.Ss1[i]
+				}
+				if i < len(test.Ss2) {
+					b = test.Ss2[i]
+				}
+
+				assert.Equal(t, pair.A, a)
+				assert.Equal(t, pair.B, b)
```

Comment:
I would prefer if the expect result was static and included in the tests. It took me a moment to figure out what was going on and I don't like tests that generate their own expectations.

Reply:
Ok, btw maybe put `ZipLongest` in the same file as `Zip`?

Reply:
The tests need to use the expected values. You can remove a lot of unnecessary code:

```go
func TestZipLongest(t *testing.T) {
 	for _, test := range zipTests {
 		t.Run("", func(t *testing.T) {
 		 	assert.Equal(t, test.expectedLong, pie.ZipLongest(test.ss1, test.ss2))
 		})
 	}
 }
```